elide-core/src/test/java/com/yahoo/elide/graphql/FetcherUpsertTest.java

```diff
@@ -0,0 +1,347 @@
+                        "id " +
+                        "books(op:UPSERT, data: [{id: \"1\", title: \"abc\"}, {id: \"2\", title: \"xyz\"}]) { " +
+                            "id " +
+                            "title " +
+                        "} " +
+                    "} " +
+                "}";
+        String expectedResponse =
+                "{" +
+                    "\"author\":[{" +
+                        "\"id\":\"1\"," +
+                        "\"books\":[{" +
+                            "\"id\":\"1\"," +
+                            "\"title\":\"abc\"" +
+                        "},{" +
+                            "\"id\":\"2\"," +
+                            "\"title\":\"xyz\"" +
+                        "}]" +
+                    "}]" +
+                "}";
+
+        assertQueryEquals(graphQLRequest, expectedResponse);
+    }
+
+    @Test
+    public void testUpsertOnCollection() {
+        String graphQLRequest =
+                "mutation { " +
+                    "author(op:UPSERT, data: {name: \"John Snow\", books: [{id: \"1\", title: \"my id\"}, {title: \"abc\"}]}) { " +
+                        "name " +
+                        "books { " +
+                            "title " +
+                        "} " +
+                    "} " +
+                "}";
+        String expectedResponse =
+                "{" +
+                    "\"author\":[{" +
+                        "\"name\":\"John Snow\"," +
+                        "\"books\":[{" +
+                            "\"title\":\"my id\"" +
+                        "},{" +
+                            "\"title\":\"abc\"" +
+                        "}]" +
+                    "}]" +
+                "}";
+        assertQueryEquals(graphQLRequest, expectedResponse);
+    }
+
+    @Test
+    public void testNonCreatedIdReferenceCollection() {
+        String graphQLRequest =
+                "mutation { " +
+                    "author(op:UPSERT, data: {id: \"1\", name: \"John Snow\", books: [{id: \"3\", title: \"updated title\"}, {title: \"abc\"}, {id: \"2\", title: \"new title\"}]}) { " +
+                        "id " +
+                        "name " +
+                        "books { " +
+                            "title " +
+                        "} " +
+                    "} " +
+                "}";
+
+        String expectedResponse =
+                "{" +
+                    "\"author\":[{" +
+                        "\"id\":\"1\"," +
+                        "\"name\":\"John Snow\"," +
+                        "\"books\":[{" +
+                            "\"title\":\"new title\"" +
+                        "},{" +
+                            "\"title\":\"updated title\"" +
+                        "},{" +
+                            "\"title\":\"abc\"" +
+                        "}]" +
+                    "}]" +
+                "}";
+        assertQueryEquals(graphQLRequest, expectedResponse);
+    }
+
```

Comment:
You should add a test case like:
```
books(op:UPSERT, data: some data containing authors)
   authors(op: UPSERT, data: some different data)
```
Verify this query works as expected (the final result is the list of authors updated by the final mutation).